addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -593,5 +631,64 @@ odoo.define('website.s_website_form', function (require) {
         _onFieldInput() {
             this._updateFieldsVisibility();
         },
+        /**
+         * Detects if files are uploaded, in order to modify the text displayed
+         * on the input file button afterwards.
+         * The modification is made by hiding the real button and then creating
+         * a fake one, along with a text displaying information about the
+         * uploaded files next to it.
+         * @private
+         * @param {MouseEvent} ev
+         */
+        _onFileChange(ev) {
+            const realButton = ev.currentTarget;
+            const parentNode = ev.target.parentNode;
+            let fakeButton = parentNode.querySelector('input[class="fake-button mr-1"]');
+            let fileInfo = parentNode.querySelector('label[class="file-info"]');
+
+            // Create a fake button with "Replace Files" as text.
+            if (!fakeButton) {
+                // Hide the current input button.
+                realButton.setAttribute('style', 'display:none');
+
+                // Create the fake button.
+                fakeButton = document.createElement('input');
+                fakeButton.type = 'button';
+                fakeButton.classList.add('fake-button');
+                fakeButton.classList.add('mr-1');
+                fakeButton.value = _t("Replace Files");
+                fakeButton.title = realButton.title;
+                parentNode.appendChild(fakeButton);
+
+                // Create the information text (empty for now).
+                fileInfo = document.createElement('label');
+                fileInfo.className += 'file-info';
+                fileInfo.style = 'font-weight:normal'; // To remove the bold effect.
```

Comment:
Given the very temporary nature of this label I guess it is fine like this. (Maybe put a space after the colon, or assign `normal` to the style's attribute directly.)
But instead you could define this rule for this label with this class in the `001.scss` related to this snippet.